node/src/components/chain_synchronizer/event.rs

```diff
@@ -10,9 +10,9 @@ use crate::{effect::requests::NodeStateRequest, types::BlockHeader};
 #[allow(clippy::enum_variant_names)]
 pub(crate) enum Event {
     /// The result of running the fast sync task.
-    FastSyncResult {
-        result: Box<Result<BlockHeader, Error>>,
-    },
+    SyncToGenesisResult(Box<Result<(), Error>>),
```

Comment:
Do we really have to box it if it's just `()` or `Error`?

Reply:
Most probably the `Error` grew big enough to start triggering the `large-enum-variant` lint. Clippy claims that `Result<(), Error>` is 272 bytes long, but `GetNodeState(NodeStateRequest)` is only 16 bytes long, hence, without boxing we can create a large memory overhead.